nixos/modules/services/misc/sonarr.nix

```diff
@@ -27,6 +36,31 @@ in
         '';
       };
 
+      settings = lib.mkOption {
+        inherit (settingsFormat) type;
+        description = "An attribute set containing Sonarr configuration settings.";
+        default = { };
```

Comment:
AFAICT, changes done in the web UI is still allowed, but will be reverted on the next service start. I think that should be noted in the `description` field. How about

> An attribute set containing Sonarr configuration written to the config.xml file, which will be applied on every service start. IOW, it overwrites imperative changes done in web UI.

I think config.xml covers roughly the stuff in Settings -> General -- should we mention it?

By reading the generated documentation, it seems the default settings are empty. But they're not. Can we have the actual default settings in the option default value?

Reply:
so overall you only suggest an documentation change? Or can we only write, when the config changes?

Reply:
I suggest to update the description _and_ wonder if we can put the default values right there in `default = ...`.

Reply:
I'm not sure what to do about the "this affects _some_ of the applications config" issue -- I was just thinking out loud.

Reply:
> I suggest to update the description and wonder if we can put the default values right there in default = ....

i have done this, because @ambroisie suggested it as well.

Reply:
@bjornfor are you now happy with the current version?

Reply:
@Stunkymonkey: Using envvars instead of overwriting config.xml seems good to me, since it aligns better with upstream (AFAICT).

But we should fix the secret leak I commented a few minutes earlier.

Reply:
resolved via 1c0d92070874ed1637e0c5c711c98f9ec973663c (will be squashed)